packages/components/FocusZone/macos/RCTFocusZone.m

```diff
@@ -362,34 +363,38 @@ - (NSView *)nextViewToFocusOutsideZone:(FocusZoneAction)action
 	NSView *nextViewToFocus;
 
 	[[self window] recalculateKeyViewLoop];
-
-	if (action == FocusZoneActionTab)  // Advance to next zone
-	{
-		IsViewLeadingCandidateForNextFocus block = ^BOOL(NSView *candidateView)
-		{
-			return YES;
-		};
-
-		nextViewToFocus = [([self nextViewToFocusForCondition:block] ?: self) nextValidKeyView];
-	}
-	else  // action == FocusZoneActionShiftTab per the conditional in keyDown
+	
+	// Find the first view outside the FocusZone (or any parent FocusZones) to place focus
+	RCTFocusZone *parentFocusZone = self;
+	RCTFocusZone *focusZoneAncestor = GetFocusZoneAncestor(parentFocusZone);
+	while (focusZoneAncestor != nil)
 	{
-		nextViewToFocus = [self previousValidKeyView];
+		parentFocusZone = focusZoneAncestor;
+		focusZoneAncestor = GetFocusZoneAncestor(parentFocusZone);
```

Comment:
I did it this way to be slightly more efficient. Line 369 calls it the first time so that the check in line 470 doesn't have to call it. Line 373 is the "looping" call. With a while loop, I figured I have to call it once, and then in a loop

Reply:
Removed this code so redundant